conf/Makefile.chibios

```diff
@@ -20,66 +19,77 @@
 # Boston, MA 02111-1307, USA.
 #
 
-
-# Verify that libopencm3 is compiled with chibios compatible float abi
-# correct abi is -mfloat-abi=softfp
-# on non chibios branches, libopencm3 is compiled with more optimised -mfloat-abi=hard
-TAG_ABI_VFP_COUNT= \
-	$(shell (readelf -A  $(PAPARAZZI_SRC)/sw/ext/libopencm3/lib/libopencm3_stm32f4.a | \
-	grep -c Tag_ABI_VFP_args))
-
-ifneq ($(TAG_ABI_VFP_COUNT),0)
-  $(error libopencm3_stm32f4.a is generated with chibios incompatible -mfloat-abi=hard, \
-	please recompile with -mfloat-abi=softfp))
-endif
-
-
 #
 # This is the common Makefile for target using chibios
+#
+# chibios directory
+CHIBIOS =  $(PAPARAZZI_SRC)/sw/ext/chibios
+
+# directory with board defines for chibios platforms (board specific)
+BOARD_DIR ?= $(BOARD)/chibios
+CHIBIOS_BOARD_DIR = $(PAPARAZZI_SRC)/sw/airborne/boards/$(BOARD_DIR)
+
+# directory with board and OS configuration (project specific)
+PROJECT_DIR ?= demo
+CHIBIOS_PROJECT_DIR = $(CHIBIOS_BOARD_DIR)/$(PROJECT_DIR)
 
-CHIBIOS_BOARD_DIR = $(PAPARAZZI_SRC)/sw/airborne/boards/$(BOARD)/chibios-libopencm3
-CHIBIOS_LIB_DIR = $(PAPARAZZI_SRC)/sw/airborne/subsystems/chibios-libopencm3
-CHIBIOS_EXT =  $(PAPARAZZI_SRC)/sw/ext/chibios
-OPENCM3_EXT =  $(PAPARAZZI_SRC)/sw/ext/libopencm3
-PPRZ_GENERATED = $(AIRCRAFT_BUILD_DIR)/$(AIRCRAFT)/$(TARGET)/generated
-PPRZ_CHIBIOS_OCM3 = $(SRC_FIRMWARE)/chibios-libopencm3
+# chibos arch directory
+CHIBIOS_ARCH_DIR = $(PAPARAZZI_SRC)/sw/airborne/arch/chibios
 
 # Launch with "make Q=''" to get full command display
 Q=@
 
+# Debugging options
+RTOS_DEBUG ?= 1
+
+ifeq (,$(findstring $(RTOS_DEBUG),0 FALSE))
+$(info DEBUGGING ON)
+$(TARGET).CFLAGS += \
```

Comment:
Not all debug options need to be set to TRUE for debugging, because then the generated files have several hundreds of megabytes. We are using this:
```
# Use only gddb2 for debugging unless otherwise specified
DEBUG_LEVEL ?= 2

ifeq (,$(findstring $(RTOS_DEBUG),0 FALSE))
$(info DEBUGGING ON)
$(info DEBUG_LEVEL = -ggdb$(DEBUG_LEVEL))
$(TARGET).CFLAGS += \
  -DCH_DBG_STATISTICS=FALSE \
	-DCH_DBG_SYSTEM_STATE_CHECK=TRUE \
	-DCH_DBG_ENABLE_CHECKS=TRUE \
	-DCH_DBG_ENABLE_ASSERTS=FALSE \
	-DCH_DBG_ENABLE_TRACE=FALSE \
	-DCH_DBG_ENABLE_STACK_CHECK=TRUE \
	-DCH_DBG_FILL_THREADS=FALSE \
	-DCH_DBG_THREADS_PROFILING=TRUE
$(TARGET).CFLAGS += -DRTOS_DEBUG=1

USE_OPT = -std=gnu99 -O0 -ggdb$(DEBUG_LEVEL) -fomit-frame-pointer -falign-functions \
	 -W -Wall -DUSE_CHIBIOS_RTOS

else
$(info DEBUGGING OFF)
$(TARGET).CFLAGS += -DCH_DBG_THREADS_PROFILING=TRUE
endif
```
Indeed it would be nice to be able to quickly set min/medium/max debugging levels - maybe merge as is, I can make a pull request with this later.